plutus-ledger-api/src/Plutus/V1/Ledger/Scripts.hs

```diff
@@ -196,17 +203,26 @@ evaluateScript s = do
             UserEvaluationError evalError -> EvaluationError logOut (PLC.show evalError)  -- TODO fix this error channel fuckery
     Haskell.pure (budget, logOut)
 
+{- Note [JSON instances for Script]
```

Comment:
FYI @j-mueller . I had to do a bit of trickery to make sure the JSON serialization which is used for the validation events *didn't* have constant size limits, otherwise it blew up due to the data constants for the context arguments, which are part of the `Script` by that point.